src/geometry/adjoints.jl

```diff
@@ -0,0 +1,52 @@
+using Base: Tuple
+function ChainRulesCore.rrule(::typeof(Interpolations.BSplineInterpolation), T, A, it, axs)
+    y = Interpolations.BSplineInterpolation(T, A, it, axs)
+    return (y, Δ -> begin
+        (NoTangent(), NoTangent(), Δ, NoTangent(), NoTangent())
+    end)
+end
```

Comment:
For constructors and functors, I don't think can dispatch like this: it should be

```julia
function ChainRulesCore.rrule(::Interpolations.BSplineInterpolation, T, A, it, axs)
```

See also: https://juliadiff.org/ChainRulesCore.jl/dev/writing_good_rules.html#Structs:-constructors-and-functors

Reply:
Stylistically, CR does suggest a `return` statement, but you don't need to wrap it in a tuple. Also you may want to write the `function $fn_to_be_differentiated_pullback`  syntax.

Reply:
> For constructors and functors, I don't think can dispatch like this: it should be

Yeah exactly, all the `rrule`s were written the same way before I started getting some errors during testing. I'll modify them again.